src/Compilers/CSharp/csc/Program.cs

```diff
@@ -8,6 +9,14 @@ namespace Microsoft.CodeAnalysis.CSharp.CommandLine
     public class Program
     {
         public static int Main(string[] args)
-            => BuildClient.RunWithConsoleOutput(args, RequestLanguage.CSharpCompile, Csc.Run);
+        {
+            var clientDir = AppDomain.CurrentDomain.BaseDirectory;
+            return BuildClient.RunWithConsoleOutput(
+                args,
+                clientDir: clientDir,
+                workingDir: Environment.CurrentDirectory,
+                language: RequestLanguage.CSharpCompile,
+                fallbackCompiler: x => Csc.Run(clientDir, x));
```

Comment:
You could supply the ```clientDir``` argument to this Func from within ```RunWithConsoleOutput``` rather than capturing it here. The ```Func<string[], int>``` parameter becomes a ```Func<string[], string, int>```.